tfjsonpath/path.go

```diff
@@ -69,34 +82,38 @@ func (s Path) AtMapKey(key string) Path {
 // is not found in the given object or if the given object does not
 // conform to format of Terraform JSON data.
 func Traverse(object any, attrPath Path) (any, error) {
-	_, ok := object.(map[string]any)
-
-	if !ok {
-		return nil, fmt.Errorf("cannot convert given object to map[string]any")
-	}
-
 	result := object
 
+	var steps []string
+
 	for _, step := range attrPath.steps {
 		switch s := step.(type) {
 		case MapStep:
+			steps = append(steps, string(s))
+
 			mapObj, ok := result.(map[string]any)
+
 			if !ok {
-				return nil, fmt.Errorf("path not found: cannot convert object at MapStep %s to map[string]any", string(s))
+				return nil, fmt.Errorf("path not found: cannot convert object at MapStep %s to map[string]any", strings.Join(steps, "."))
 			}
+
 			result, ok = mapObj[string(s)]
+
 			if !ok {
-				return nil, fmt.Errorf("path not found: specified key %s not found in map", string(s))
+				return nil, fmt.Errorf("path not found: specified key %s not found in map at %s", string(s), strings.Join(steps, "."))
 			}
 
 		case SliceStep:
+			steps = append(steps, fmt.Sprint(s))
```

Comment:
Is there a use case for using `fmt.Sprint(s)` instead of `string(s)` here?

Reply:
There's no use-case for `fmt.Sprint(s)`, have updated to `string(s)`.